src/agent/windows/CMDCommunication.cpp

```diff
@@ -34,9 +48,31 @@ std::string CMDCommunication::ExecuteDiscStatusCommand(const Disk& _disk) const
 
     std::string ret = output.toStdString();
     
-    auto diskPos = ret.find(_disk.deviceId());
-    auto statusPosStart = ret.find_first_not_of(' ', diskPos + (_disk.deviceId()).size());
+    auto diskPos = ret.find(_disk.GetDeviceId());
+    auto statusPosStart = ret.find_first_not_of(' ', diskPos + (_disk.GetDeviceId()).size());
+    auto statusPosStop = ret.find_first_of(' ', statusPosStart);
+    ret = ret.substr(statusPosStart, statusPosStop - statusPosStart);
+    return ret;
+}
+
+std::string CMDCommunication::GetInstanceName(const Disk& _disk) const
+{
+    QProcess proc;
+    proc.start("wmic diskdrive get DeviceID, PNPDeviceID");
+    proc.waitForFinished();
+    QString output = proc.readAllStandardOutput();
+
+    std::string ret = output.toStdString();
+
+    auto diskPos = ret.find(_disk.GetDeviceId());
+    auto statusPosStart = ret.find_first_not_of(' ', diskPos + (_disk.GetDeviceId()).size());
     auto statusPosStop = ret.find_first_of(' ', statusPosStart);
     ret = ret.substr(statusPosStart, statusPosStop - statusPosStart);
     return ret;
-}
\ No newline at end of file
+}
+
+void CMDCommunication::ChangeStringToLowerCase(std::string& _string) const
+{
+    std::transform(_string.begin(), _string.end(), _string.begin(),
+        [](unsigned char c) { return std::tolower(c); });

```

Comment:
wouldn't std::tolower instead of lambda work?

Reply:
Would it give me any advantage?

Reply:
I belive we met such situation few PRs ago. And it won't.

Reply:
something like this: std::transform(_string.begin(), _string.end(), _string.begin(), std::tolower);